File: server/php/src/Presentation/Models/HostedTokenization/Response.php
<?php

namespace OnlinePayments\ExampleApp\Presentation\Models\HostedTokenization;

use OnlinePayments\ExampleApp\Application\Interfaces\Responses\ResponseInterface;

class Response implements ResponseInterface
{
    public ?string $hostedTokenizationId = null;
    public ?string $hostedTokenizationUrl = null;

    public function __construct(?string $hostedTokenizationId = null, ?string $hostedTokenizationUrl = null)
    {
        $this->hostedTokenizationId = $hostedTokenizationId;
        $this->hostedTokenizationUrl = $hostedTokenizationUrl;
    }

    public function toArray(): array
    {
        return [
            'hostedTokenizationId' => $this->hostedTokenizationId,
            'hostedTokenizationUrl' => $this->hostedTokenizationUrl,
        ];
    }
}

File: server/ruby/app/presentation/models/hosted_tokenization/response.rb
module Presentation
  module Models
    module HostedTokenization
      class Response
        attr_reader :hosted_tokenization_id, :hosted_tokenization_url

        def initialize(hosted_tokenization_id:, hosted_tokenization_url:)
          @hosted_tokenization_id  = hosted_tokenization_id
          @hosted_tokenization_url = hosted_tokenization_url
        end

        def as_json(*)
          {
            hostedTokenizationId: hosted_tokenization_id,
            hostedTokenizationUrl: hosted_tokenization_url
          }
        end
      end
    end
  end
end
